Fixs error bar shown with valid owners

orelbn opened the pull request:
*Issue #:* /bcgov/entity#16395

*Description of changes:*
- Removes overlapping `showTableError` conditions
- Updates validation to allow for no groups shown and 2+ owners of type administrator/bankrupcy trustee/executor

*Screenshots:*

- Before change when returning from review and confirm
![image](https://github.com/bcgov/ppr/assets/77707952/9bdda0b3-a009-47ef-8ead-0b9982fa18ff)


- After change when returning from review and confirm 
![image](https://github.com/bcgov/ppr/assets/77707952/23305a12-8377-4675-af9d-50bd6550fa8f)


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

Commit:
Updates validation to allow noGroups & N/A teancy

Commit:
version bump

orelbn commented:
/gcbrun

orelbn commented on ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue:
```diff
@@ -512,17 +509,13 @@ export default defineComponent({
         return ((props.validateTransfer || (!props.isMhrTransfer && localState.reviewedOwners)) &&
           (
             !hasMinimumGroups() ||
-            hasEmptyGroup.value ||
```
`localState.hasGroupsWithNoOwners` checks there is non-empty groups.

orelbn commented on ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue:
```diff
@@ -512,17 +509,13 @@ export default defineComponent({
             (props.isMhrTransfer && !hasUnsavedChanges.value) ||
             !localState.isValidAllocation ||
             localState.hasGroupsWithNoOwners ||
-            (!localState.isUngroupedTenancy && hasUndefinedGroupInterest(groups)) ||
```
`isUngroupedTenancy` checked that the tenancy type is sole owner or joints tenants, however we can have an N/A tenancy type, and no explicit groups. Under the condition above we will return true (showTableError) because the group interest will be undefined.

orelbn replied:
`isValidAllocation`  when showing groups checks that for non-removed groups the group interest is defined.

dimak1 replied:
What about `hasUndefinedGroupInterest`? Where did that go?

orelbn replied:
`isValidAllocation`, if `showGroups.value` is true checks for total allocation error, which covers `hasUndefiendedGroupInterest`

![image](https://github.com/bcgov/ppr/assets/77707952/f672dea1-8133-4a10-b867-e00fffed7de9)

![image](https://github.com/bcgov/ppr/assets/77707952/b191b34e-6ebc-4fee-a3e7-41074afe30e7)

dimak1 replied:
ok, got it, thanks

orelbn commented on ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts:
```diff
@@ -189,8 +189,8 @@ export function useHomeOwners (isMhrTransfer: boolean = false) {
     const groups = getTransferOrRegistrationHomeOwnerGroups().filter(group => group.action !== ActionTypes.REMOVED)
-    const hasNoGroups = [HomeTenancyTypes.SOLE, HomeTenancyTypes.JOINT].includes(getHomeTenancyType())
-    return hasNoGroups || !groups || groups.length >= 2
+    return !groups || groups.length >= 2 ||
```
I am not sure why we have !groups condition here. If anyone knows?
I just kept it since I was not certain if we need it or not.

dimak1 replied:
Not sure why we have !groups, but there are many scenarios for validations, so I assume at some point it was needed.

pwei1018 commented:
Temporary Url for review: https://bcregistry-assets-dev--pr-1349-e243rf6j.web.app

dimak1 reviewed (commented):
Could you please also test Transfers (if not done so) to make sure everything works as usual? Any updates to validations affects many flows, and would require a regression, from QA as well...

orelbn commented:
> Could you please also test Transfers (if not done so) to make sure everything works as usual? Any updates to validations affects many flows, and would require a regression, from QA as well...

Ill test transfers and post screenshots.

I tested transfer and was not able to identify differences in validation between the pr temporary link and dev environment.

1.Transfer due to sale or gift validations (after hitting review and confirm)

DEV:

No Owners: 

![dev-1](https://github.com/bcgov/ppr/assets/77707952/353ac1ca-4848-4c6a-ba85-5dfa8be8430f)

![dev-2](https://github.com/bcgov/ppr/assets/77707952/21086703-1d02-4aa8-9e99-f4c5432de64d)

Invalid Allocation:

![dev-transfer-due-to-sale-or-gift](https://github.com/bcgov/ppr/assets/77707952/93622c82-c30a-4208-b518-7efd19230af1)

Removed Groups:

![dev-transfer-due-to-sale-or-gift-2](https://github.com/bcgov/ppr/assets/77707952/975f655e-f7f2-46f5-b246-574d60da0895)



PR:

No Owners: 

![pr-1](https://github.com/bcgov/ppr/assets/77707952/c20f48a3-1699-4f94-a57c-7004fadc5eef)

![pr-2](https://github.com/bcgov/ppr/assets/77707952/0c9f5fac-b0ae-4552-be47-2f47a51501b8)

Invalid Allocation:

![pr-trasnsfer-due-to-sale-or-gift](https://github.com/bcgov/ppr/assets/77707952/21e3b877-0efe-4065-b729-18b52d77b091)

Removed Groups:

![pr-trasnsfer-due-to-sale-or-gift-2](https://github.com/bcgov/ppr/assets/77707952/b198c0b2-54b4-49d4-8409-d5f8ec110a36)



* *Groups can not be updated in the following transfer types *

2. Transfer to Surviving Joint Tenant(s)

DEV:

No owners:

![dev-tsjt-2](https://github.com/bcgov/ppr/assets/77707952/9c88af5a-15ad-4037-b75c-cac089a9da12)

No changes:

![dev-tsjt-1](https://github.com/bcgov/ppr/assets/77707952/b85a9cd0-95f1-4dc2-ae19-879dc0922b3c)


PR:

No owners:

![pr-tsjt-2](https://github.com/bcgov/ppr/assets/77707952/0398af74-16af-44de-be18-51dace99b1c0)

No changes:

![pr-tsjt-1](https://github.com/bcgov/ppr/assets/77707952/f935194e-4405-4eec-8240-7c6275eb8462)

3. Transfer Grant/Wil Admin/Executor:

DEV:

Removed owners:

![dev-grant-wth-will-2](https://github.com/bcgov/ppr/assets/77707952/08e4c86a-ddf9-4da5-a236-bb8e7b71da85)

PR:

Removed owners:

![pr-grant-with-will-2](https://github.com/bcgov/ppr/assets/77707952/e85e9947-86c8-44a1-92be-36da41f99657)

![pr-exe-group](https://github.com/bcgov/ppr/assets/77707952/4d028d07-f537-424e-b22b-e5565046582a)

Extra validation assurance in PR:

![pr-exe-group-2](https://github.com/bcgov/ppr/assets/77707952/7a9e2ccf-ad22-4d95-ba83-c2e4e046c35e)

![group-admin-1](https://github.com/bcgov/ppr/assets/77707952/727523bd-afec-4200-95c2-be1ce7d298e3)

![group-admin-2](https://github.com/bcgov/ppr/assets/77707952/2874e6c4-c19e-4a65-9026-e6963395c4f4)

doug-lovett reviewed (approved):
LGTM.

dimak1 commented:
@orelbn thanks for putting the work in to test the Transfers. You said there are no differences between PR and DEV? I assume that's besides the changes you made? In that case, your PR just fixed the bug and the rest works as expected. Right?

orelbn commented:
> @orelbn thanks for putting the work in to test the Transfers. You said there are no differences between PR and DEV? I assume that's besides the changes you made? In that case, your PR just fixed the bug and the rest works as expected. Right?

Yes, **from my review**, except fixing the bug, the rest works as it did prior to the changes.

If you find anything that I missed, please let me know.
